include/fiction/algorithms/network_transformation/inverter_balancing.hpp

```diff
@@ -0,0 +1,320 @@
+                    if (ntk.is_and(g))
+                    {
+                        old2new[g] = ntk_dest.create_and(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_or_v<TopoNtkSrc> && mockturtle::has_create_or_v<Ntk>)
+                {
+                    if (ntk.is_or(g))
+                    {
+                        old2new[g] = ntk_dest.create_or(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_xor_v<TopoNtkSrc> && mockturtle::has_create_xor_v<Ntk>)
+                {
+                    if (ntk.is_xor(g))
+                    {
+                        old2new[g] = ntk_dest.create_xor(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_maj_v<TopoNtkSrc> && mockturtle::has_create_maj_v<Ntk>)
+                {
+                    if (ntk.is_maj(g))
+                    {
+                        old2new[g] = ntk_dest.create_maj(children[0], children[1], children[2]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_nary_and_v<TopoNtkSrc> && mockturtle::has_create_nary_and_v<Ntk>)
+                {
+                    if (ntk.is_nary_and(g))
+                    {
+                        old2new[g] = ntk_dest.create_nary_and(children);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_nary_or_v<TopoNtkSrc> && mockturtle::has_create_nary_or_v<Ntk>)
+                {
+                    if (ntk.is_nary_or(g))
+                    {
+                        old2new[g] = ntk_dest.create_nary_or(children);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_nary_xor_v<TopoNtkSrc> && mockturtle::has_create_nary_xor_v<Ntk>)
+                {
+                    if (ntk.is_nary_xor(g))
+                    {
+                        old2new[g] = ntk_dest.create_nary_xor(children);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_node_function_v<TopoNtkSrc> && mockturtle::has_create_node_v<Ntk>)
+                {
+                    old2new[g] = ntk_dest.create_node(children, ntk.node_function(g));
+                    return true;
+                }
+
+                return true;
+            });
+
+        ntk.foreach_po(
+            [this, &ntk_dest, &old2new](const auto& po)
+            {
+                auto tgt_signal = old2new[ntk.get_node(po)];
+                auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+                if (tgt_po == 0)
```

Comment:
You're comparing a `mockturtle::signal<Ntk>` with an `int`. That can work, but there is no guarantee that the former has to be a numerical type or offer such an equality operator. Maybe we can replace `0` with `mockturtle::signal<Ntk>{}`?